src/components/ListingTable.vue

```diff
@@ -122,7 +122,29 @@ export default {
   name: "ListingTable",
   props: ["listings"],
   setup() {
-    function viewListing(id) {
+    async function viewListing(id) {
+      await fetch(`${process.env.SERVER_URL}/modify-statitics/${id}`, {
+        method: "PUT",
+        mode: "cors",
+        credentials: "same-origin",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: {"statistic":"views"},
+      })
+        .then((res) => {
+          if (res.status === 200) {
+            // Need refactor, better way of routing
+            window.location.href = "/admin";
+          } else if (res.status === 403) {
+            res.json().then((r) => {
+              alert(r.err_msg);
+            });
+          }
+        })
+        .catch((err) => {
+          console.log(err);
+        });
       window.location.href = `/view-listing/${id}`;
```

Comment:
This would be the correct route but we only want to redirect them if we have a 200 status code from the server, if we redirect them without the status code we will let them view the listing without collecting the data that they viewed it. Once again, this can be modified later down the line this sprint once I implement the view-listing page